pkg/resourcemanager/controller/managedresource/merger.go

```diff
@@ -202,9 +206,36 @@ func mergeJob(scheme *runtime.Scheme, oldObj, newObj runtime.Object) error {
 	// Do not overwrite Job managed labels as 'controller-uid' and 'job-name'. '.spec.template' is immutable.
 	newJob.Spec.Template.Labels = labels.Merge(oldJob.Spec.Template.Labels, newJob.Spec.Template.Labels)
 
+	mergePodTemplate(&oldJob.Spec.Template, &newJob.Spec.Template, preserveResources)
+
 	return scheme.Convert(newJob, newObj, nil)
 }
 
+func mergeCronJob(scheme *runtime.Scheme, oldObj, newObj runtime.Object, preserveResources bool) error {
+	oldCronJob := &batchv1.CronJob{}
+	if err := scheme.Convert(oldObj, oldCronJob, nil); err != nil {
+		return err
+	}
+
+	newCronJob := &batchv1.CronJob{}
+	if err := scheme.Convert(newObj, newCronJob, nil); err != nil {
+		return err
+	}
+
+	// Do not overwrite a CronJob's '.spec.jobTemplate.spec.selector' if the new CronJobs's '.spec.jobTemplate.spec.selector'
+	// field is unset.
+	if newCronJob.Spec.JobTemplate.Spec.Selector == nil && oldCronJob.Spec.JobTemplate.Spec.Selector != nil {
+		newCronJob.Spec.JobTemplate.Spec.Selector = oldCronJob.Spec.JobTemplate.Spec.Selector
+	}
+
+	// Do not overwrite CronJob managed labels as 'controller-uid' and 'job-name'. '.spec.jobTemplate.spec.template' is immutable.
```

Comment:
> '.spec.jobTemplate.spec.template' is immutable.

Can you explain. To a certain degree it seems to be mutable because one can for instance add labels, right? Furthermore, I don't see any validation blocking those updates https://github.com/kubernetes/kubernetes/blob/c16b2afc1d3c32462f068ea08cdc4791bd97b947/pkg/apis/batch/validation/validation.go#L296. Was this the case in previous versions?

Reply:
I guess this handling was copied from the `mergeJob` func. For the Job resource I can confirm that its `spec.template` is immutable - ref https://github.com/kubernetes/kubernetes/blob/c16b2afc1d3c32462f068ea08cdc4791bd97b947/pkg/apis/batch/validation/validation.go#L239 and https://github.com/kubernetes/kubernetes/blob/c16b2afc1d3c32462f068ea08cdc4791bd97b947/pkg/apis/batch/validation/validation.go#L269

Looks like similar validation is not present for the CronJob.

Reply:
Thanks folks, I dropped this code accordingly.